erpnext/accounts/doctype/shipping_rule/shipping_rule.py

```diff
@@ -113,7 +113,31 @@ def add_shipping_rule_to_tax_table(self, doc, shipping_amount):
 		else:
 			shipping_charge["tax_amount"] = shipping_amount
 			shipping_charge["description"] = self.label
-			doc.append("taxes", shipping_charge)
+
+			# Check if already tax is presnt & add new shipping taxes
+			if doc.get("taxes"):
+				for i, tax in enumerate(doc.get("taxes")):
+					tax = tax.as_dict()
+
+				doc.set("taxes", [])
+				doc.append("taxes", shipping_charge)
+				doc.append("taxes", tax)
+
+			else:
+				doc.append("taxes", shipping_charge)
+
+	# remove existing shipping rule tax by its account head from taxes table
+	def remove_existing_shipping_rule(self, doc, doctype, docname):
+		if not docname:
+			return
+
+		shipping_rule_accounts = frappe.get_all("Shipping Rule", fields={"account"})
```

Comment:
- fields should be a `list`.

Reply:
As per our discussion, we are deleting the records by its account head only, that's why the records fetched by the account head

Reply:
- I am saying fields should be a `list` not `dict`